src/libdredd/src/mutation_replace_expr.cc

```diff
@@ -62,23 +62,39 @@ std::string MutationReplaceExpr::GetFunctionName(
     result += "_lvalue";
   }
 
+  if (IsRedundantOperatorInsertion(expr_, ast_context)) {
+    result += "_optimised";
+  }
```

Comment:
This is fine if there's only one way that the generated function changes due to redundancy. If there are are multiple different ways then we will get name clashes. Can you please check? If it's fine as is, please add a reassuring comment.

Reply:
This does appear to be fine because this is the only optimization that depends on the expression. The other UOI optimization that is applied, omitting `~`, is dependent on the expression type being bool, so the type included in the function name should be enough to distinguish this case.